frontend/src/main/scala/bloop/engine/tasks/CompileTask.scala

```diff
@@ -459,14 +459,14 @@ object CompileTask {
 
   /**
    * Prepares a clean up task that will delete the previous used read-only
-   * classes directory used as the read-only directory of an effectful compile.
+   * classes directory used as the read-only directory of an effective compile.
```

Comment:
I'm not at all certain about this.

Reply:
This one should actually be `effectful` instead of `effective`. Effectful is an adjective used to mean some computation that side-effects. Maybe we could replace it by side-effecting instead.

Reply:
I'd like to replace it with something, but I think I need more information (I'm not using bloop right now, we're merely investigating a number of technologies and one of the things I do is investigate their ecosystems).

It sounds like you have a source directory, and an objdir (to avoid dirtying the source dir), and then you take that objdir and toss it into the classpath while compiling a new objdir.

And here you're merely talking about that intermediate directory.

There are other issues w/ this paragraph:
* I suspect that `previous` should be `previously`.

* The right hand side of the either-or here is flawed:
> If a compilation is a no-op or its used counter is bigger than zero, the
> deletion is skipped because in both scenarios the classes directory is
> still useful for either a next compile or is being used by an alternative
> running compilation process.

It should be possible to read it as:

> If a compilation is a no-op or its used counter is bigger than zero, the
> deletion is skipped because in both scenarios the classes directory is
> still useful for ... is being used by an alternative
> running compilation process.

But that's clearly nonsensical. (Also, the left hand side of that either-or isn't much better.)

For now, I'm going to take the pragmatic approach and drop the commit entirely. That will leave this review comment thread orphaned, but someone can still fix it. They just need to spend real time on it, and that's inappropriate for a spelling PR.

Reply:
For the record, this commit is now dropped. I've moved the problem to #1075

Reply:
Sorry, just saw this message. What you've done in that PR looks good to me, thanks! @jsoref Let me know if you have any questions regarding Bloop as a technology. Feel free to drop by our Discord channel https://discord.gg/vys6d43 and ask questions